### DISCUSSION TITLE
Payload Admin Sign-In

### DISCUSSION THREAD

**Discussion by Jelly-exe** [CrawlerCode/payload-authjs#10] [Q&A]
Couple questions, 

I'm using this plugin to do OAuth sign in for the admin panel, I only want users with existing admin accounts to be able to use the OAuth to login. (Created by another admin manually)

- Is there a way to restrict it to only signing in for existing accounts?
- Is there a way to manually create a link, or to specify that an account with Id equal to the oauth account id will link.
- Is there a way to edit the sign in button, ideally, I'd like it to just be `Sign in with <service>`

**Comment by CrawlerCode** [comment]
**Is there a way to restrict it to only signing in for existing accounts?**

I am not currently aware of a suitable way to implement this.
You could restrict access with the [SignIn callback](https://authjs.dev/guides/restricting-user-access), but you can't check if the user exists or not. There is currently an [open issue](https://github.com/nextauthjs/next-auth/issues/8272) on authjs/next-auth that focus your problem.

---

**Is there a way to manually create a link, or to specify that an account with Id equal to the oauth account id will link.**

Could you clarify your question or give an example?
You could create a database entry (user, account or whatever) directly via the payload api.

---

**Is there a way to edit the sign in button, ideally, I'd like it to just be Sign in with <service>**

This is not possible at the moment. I will add the customization in the future.

**Comment by CrawlerCode** [comment]
By the way, I have added the option to change the text of the sign in button:

```
authjsPlugin({
  authjsConfig: authConfig,
  // Customize the sign-in button text
  components: {
    SignInButton: {
      text: "Sign in with <service>",
    },
  },
})
```
